src/World.js

```diff
@@ -311,29 +312,32 @@ class World {
 	// accepts array of notations eg: ['4d6','2d10']
 	// accepts array of objects eg: [{sides:int, qty:int, mods:[]}]
 	// accepts object {sides:int, qty:int}
-	createNotationArray(notation){
+	createNotationArray(input){
+		const notation = Array.isArray( input ) ? input : [ input ]
 		let parsedNotation = []
 
-		if(typeof notation === 'string') {
-			parsedNotation.push(this.parse(notation))
+		const verifyObject = ( object ) => {
+			const checkSidesAndQty =  object.sides && object.qty ? true : false
+	
+			if ( checkSidesAndQty ) {
+				return true
+			} else {
+				const err = "Roll notation is missing sides and/or qty"
+				throw new Error(err);
+			}
 		}
+	
 
 		// notation is an array of strings or objects
-		if(Array.isArray(notation)) {
-			notation.forEach(roll => {
-				// if notation is an array of strings
-				if(typeof roll === 'string') {
-					parsedNotation.push(this.parse(roll))
-				}
-				else {
-					// TODO: ensure that there is a 'sides' and 'qty' value on the object - required for making a roll
-					parsedNotation.push(roll)
-				}
-			})
-		} else if(typeof notation === 'object'){
-			// TODO: ensure that there is a 'sides' and 'qty' value on the object - required for making a roll
-			parsedNotation.push(notation)
-		}
+		notation.forEach(roll => {
+			// if notation is an array of strings
+			if ( typeof roll === 'string' ) {
+				parsedNotation.push( parse( roll ) )
```

Comment:
parse is undefined. I think you mean `this.parse`

Reply:
Yep. Fixed! You can tell I am well out of practice with working in classes.